rediscluster/pipeline.py

```diff
@@ -534,22 +369,45 @@ def inner(*args, **kwargs):
 StrictClusterPipeline.time = block_pipeline_command(StrictRedis.time)
 
 
-class ThreadedPipelineExecute(threading.Thread):
-    """
-    Threaded pipeline execution.
-    release the connection back into the pool after executing.
-    """
+class PipelineCommand(object):
+    def __init__(self, args, options=None, position=None):
+        self.args = args
+        if options is None:
+            options = {}
+        self.options = options
+        self.position = position
+        self.result = None
+        self.node = None
+        self.asking = False
+
+
+class NodeCommands(object):
+
+    def __init__(self, parse_response, connection):
+        self.parse_response = parse_response
+        self.connection = connection
+        self.commands = []
 
-    def __init__(self, execute, conn, cmds):
-        threading.Thread.__init__(self)
-        self.execute = execute
-        self.conn = conn
-        self.cmds = cmds
-        self.exception = None
-        self.value = None
+    def append(self, c):
+        self.commands.append(c)
 
-    def run(self):
+    def write(self):
+        """
+        Code borrowed from StrictRedis so it can be fixed
+        """
+        # build up all commands into a single request to increase network perf
+        connection = self.connection
+        commands = self.commands
         try:
-            self.value = self.execute(self.conn, self.cmds, False)
-        except Exception as e:
-            self.exception = e
+            connection.send_packed_command(connection.pack_commands([c.args for c in commands]))
+        except ConnectionError as e:
+            for c in commands:
+                c.result = e
+
+    def read(self):
+        connection = self.connection
```

Comment:
I did it as an optimization for the loop iteration. local variables are faster to access. But really it probably doesn't matter. Do you have a strong preference?